src/rustdoc-cache.ts

```diff
@@ -73,9 +73,15 @@ export class RustdocCache {
     private getRunDependentKey(): string {
         return [
             process.env["GITHUB_JOB"] || "",
-            rustCore.input.getInput("package"),
+            "package",
+            rustCore.input.getInputList("package").sort(),
```

Comment:
A bit of a nitpick, but let's be a bit more clever here to avoid a potentially difficult-to-debug situation later.

Here's what can happen right now: `package: foo-bar` and `package: foo, bar` produce the same cache key, and can overwrite each other's cache which will lead to poor performance.

The easiest way to do better here is to observe that at some point, this key name just grows too long and unwieldy to be useful, and then:
- use a format that is resistant to the confusion I described above
- hash the representation in that format, and just use the hash as the cache key suffix after the `GITHUB_JOB` component.

I'd recommend using three JSON objects:
```
{
    "manifest-path": "path"
}

{
    "package": [ // sorted list ]
}

{
    "exclude": [ // sorted list ]
}
```
Why three objects? So that we don't have to worry about whether keys get sorted differently -- one key per object means no ambiguity.

Then make a hasher, feed the JSON representations into the hasher in this order (remember order matters), and use the first 16 hex chars of the hash as the cache key component. 16 hex chars is 8 bytes so the random collision chance even with the birthday paradox in play would be one in billions.

Reply:
This is a great idea, thank you! I'll implement this in a moment